design/upgrade.md

```diff
@@ -0,0 +1,165 @@
+# **Rook Cluster Upgrades**
+
+## **Overview**
+Over time, new versions with improvements to the Rook software will be released and Rook clusters that have already been deployed should be upgraded to the newly released version.
+Being able to keep the deployed software current is an important part of managing the deployment and ensuring its health.
+In the theme of Rook's orchestration and management capabilities making the life of storage admins easier, this upgrade process should be both automatic and reliable.
+This document will describe a proposed design for the upgrading of Rook software as well as pose questions to the community for feedback so that we can deliver on the goal of automatic and reliable upgrades.
+
+## **Goal**
+In order for software upgrade support in Rook to be considered successful, the goals listed below should be met.
+Note that these goals are for a long term vision and are not all necessarily deliverable within the v0.6 release time frame.
+* **Automatic:** When a new version of Rook is released, a live cluster should be able to update all its components to the new version without user intervention.
```

Comment:
I dont think we should do `automatic` upgrade. Admins likes to be in control and doing an upgrade silently at the wrong time could impact our users and make admins not happy. I think the upgrade should be initialize by the admin.

Reply:
I would agree that all the automation happens *after* the admin has decided that he is ready to install the new version. But it looks like that's how the update steps are defined below.

Reply:
i'll clarify that the upgrade process should be fully automated AFTER the user has chosen to start it and it will complete without further user intervention.